frontend/src/metabase/query_builder/utils.unit.spec.ts

```diff
@@ -9,201 +9,274 @@ import { checkNotNull } from "metabase/core/utils/types";
 
 import { isNavigationAllowed } from "./utils";
 
-const mockCard = createMockCard({
+const notebookCard = createMockCard({
```

Comment:
Summary of changes in this test file:
- remove `mock` prefix from identifiers (everything is a mock here)
- add `notebook` prefix to `card` and `question` (to indicate how the question was created)
- add `name` to cards to be used in `it.each`
- add test for the new behavior (creating a model)
- use `it.each` & `describe.each` where possible to ease debugging
- drop `locations` array and provide as exhaustive list of locations as possible to each `it.each`
- remove duplicated tests for undefined `question` and `destination`